torch/_refs/fft.py

```diff
@@ -533,15 +550,29 @@ def hfft2(
     return torch.fft.hfftn(input, s=s, dim=dim, norm=norm)
 
 
+# DO NOT use out_wrapper and DO NOT dispatch to C++ here! Previously, this ref
+# used out_wrapper and called torch.fft.ihfftn, which dispatches to C++. This
+# wasn't correct with respect to handling conjugation since the C++ code has
+# conditional logic for that. In fft_ihfftn_impl, out.defined() is called to
+# check whether to conjugate or to just set the conj bit. Without an explicit
+# out argument passed from this ref to the C++ code, out.defined() on the C++
+# side would always return False. That would be incorrect when this ref is
+# called with an out parameter. While passing an explicit out parameter to
+# torch.fft.ihfftn works to resolve the out.defined() issue, it breaks out tests
+# for this ref, like test_out__refs_fft_ihfft2_cpu_float32. Instead, this ref
+# now calls the ihfftn ref since it resolves both of the mentioned problems.
+# The ihfftn ref implements the same conditional logic as the C++ code and
+# doesn't cause issues with out tests.
 @register_decomposition(aten.fft_ihfft2)
-@out_wrapper()
 def ihfft2(
     input: TensorLikeType,
     s: Optional[ShapeType] = None,
     dim: Optional[DimsType] = (-2, -1),
     norm: NormType = None,
+    *,
+    out: TensorLikeType = None,
 ) -> TensorLikeType:
-    return torch.fft.ihfftn(input, s=s, dim=dim, norm=norm)
+    return ihfftn(input, s=s, dim=dim, norm=norm, out=out)
```

Comment:
Couldn't use `out_wrapper` here because it calls into C++ where an explicit `out` param is required due to conditional logic for conj. But that also didn't quite work and broke `test_out__refs_fft_ihfft2_cpu_float32`. So I just ended up calling a ref instead, which has similar code as in core. See the comment for details.